tests/components/sharkiq/test_shark_iq.py

```diff
@@ -0,0 +1,283 @@
+async def test_shark_metadata(hass: HomeAssistant) -> None:
+    """Test shark properties coming from metadata."""
+    ayla_api = MockAyla()
+    shark_vac = _get_mock_shark_vac(ayla_api)
+    coordinator = SharkIqUpdateCoordinator(hass, None, ayla_api, [shark_vac])
+    shark = SharkVacuumEntity(shark_vac, coordinator)
+    shark.sharkiq._update_metadata(  # pylint: disable=protected-access
+        SHARK_METADATA_DICT
+    )
+
+    target_device_info = {
+        "identifiers": {("sharkiq", "AC000Wxxxxxxxxx")},
+        "name": "Sharknado",
+        "manufacturer": "Shark",
+        "model": "RV1001AE",
+        "sw_version": "Dummy Firmware 1.0",
+    }
+    state_json = json.dumps(shark.device_info, sort_keys=True)
+    target_json = json.dumps(target_device_info, sort_keys=True)
+    assert state_json == target_json
+
+
+def _get_async_update(err=None):
+    async def _async_update(_) -> bool:
+        if err is not None:
+            raise err
+        return True
+
+    return _async_update
+
+
+@patch.object(AylaApi, "async_list_devices", new=_async_list_devices)
+async def test_updates(hass: HomeAssistant) -> None:
+    """Test the update coordinator update functions."""
+    ayla_api = get_ayla_api(TEST_USERNAME, TEST_PASSWORD)
+    shark_vac = _get_mock_shark_vac(ayla_api)
+    mock_config = MagicMock(spec=ConfigEntry)
+    coordinator = SharkIqUpdateCoordinator(hass, mock_config, ayla_api, [shark_vac])
+
+    with patch.object(SharkIqVacuum, "async_update", new=_get_async_update()):
+        update_called = (
+            await coordinator._async_update_data()  # pylint: disable=protected-access
+        )
+    assert update_called
+
+    update_failed = False
+    with patch.object(
+        SharkIqVacuum, "async_update", new=_get_async_update(SharkIqAuthError)
+    ), patch.object(HomeAssistant, "async_create_task"), patch.object(
+        ConfigEntriesFlowManager, "async_init"
+    ):
+        try:
+            await coordinator._async_update_data()  # pylint: disable=protected-access
+        except UpdateFailed:
+            update_failed = True
+    assert update_failed
+
+
+async def test_coordinator_match(hass: HomeAssistant):
+    """Test that sharkiq-coordinator references work."""
+    ayla_api = get_ayla_api(TEST_PASSWORD, TEST_USERNAME)
+    shark_vac1 = _get_mock_shark_vac(ayla_api)
+    shark_vac2 = _get_mock_shark_vac(ayla_api)
+    shark_vac2._dsn = "FOOBAR!"  # pylint: disable=protected-access
+
+    coordinator = SharkIqUpdateCoordinator(hass, None, ayla_api, [shark_vac1])
+
+    # The first should succeed, the second should fail
+    api1 = SharkVacuumEntity(shark_vac1, coordinator)
+    try:
+        _ = SharkVacuumEntity(shark_vac2, coordinator)
+    except RuntimeError:
+        api2_failed = True
+    else:
+        api2_failed = False
+    assert api2_failed
+
+    coordinator.last_update_success = True
+    coordinator._online_dsns = set()  # pylint: disable=protected-access
+    assert not api1.is_online
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6090606)